File: Java Interfaces/src/ArrayList.java
public class ArrayList implements List {

    public int length = 0;
    public int array [] = new int[50];

    public void printArray() {
        for(int i = 0; i < this.size(); i++){
            System.out.print(array[i] + " ");
        }
    }

    @Override
    public int size() {
        return this.length;
    }

    @Override
    public void push(int value) {
        this.array[this.size()] = value;
        length++;
    }

    @Override
    public void pop() {
        int delete = array[this.size()];
        array[this.size()] = 0;
        length--;
        //System.out.println(delete);
    }

    @Override
    public void unshift(int value) {
        for(int i = this.size(); i >  0 ; i--){
            this.array[i] = this.array[i - 1];
        }
        array[0] = value;
        length++;
    }

    @Override
    public void shift() {
        String delete = this.size() + " deleted";
        for(int i = 0; i < size() ; i++){
            this.array[i] = this.array[i + 1];
        }
        length--;
        //System.out.println(delete);
    }

    @Override
    public void toString(int[] array) {
        String str = "";
        for(int i = 0; i < size() ; i++){
            str += this.array[i] + ", ";
        }
        str =  str.substring(0, str.length() - 2) + " - String";
        System.out.println(str);
    }

    @Override
    public int[] sort(int[] arr) {
        for (int i = 0, endI = size() - 1; i < endI; i++){
            boolean isSwap = false;
            int tmp;
            for (int j = 0, endJ = endI - i; j < endJ; j++){
                if (arr[j] > arr[j + 1]){
                    tmp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = tmp;
                    isSwap = true;
                }
            }
            if (!isSwap){break;}
        }
        return arr;
    }

}

File: Js Interfaces/ES5/js/ArrayList.js
function ArrayList() {
	List.call(this);
	this.size = 0;
	this.array = [];
	this.str = "";

	this.clear = function () {
		if(arguments.length > 0){
			return false;
		}
		this.array = [];
		this.size = 0;
	};
	
	this.length = function () {
		return this.size;
	};
	
	this.push = function (value) {
		if(arguments.length !== 1 || typeof(arguments) == null || typeof(arguments) == "undefined"){
			return false;
		}
		this.array[this.size] = value;
		this.size++;
		return value;
	};
	
	this.pop = function (){
		lastElem = this.array[this.size - 1];
		let a = [];
		for(let i = 0; i < this.size - 1; i++){
			a[i] = this.array[i];
		}
		this.array = a;
		this.size--;
		return lastElem;
	};
	
	this.unshift = function  (value){
		if(arguments.length !== 1){
			return false;
		}
		let a = [];
		a[0] = value;
		for(let i = 0; i < this.size; i++){
			a[i + 1] = this.array[i];
		}
		this.array = a;
		this.size++;
		return value;
	};
	
	this.shift = function () {
		firstElem = this.array[0];
		let a = [];
		for(let i = 0; i < this.size - 1; i++){
			a[i] = this.array[i + 1];
		}
		this.array = a;
		this.size--;
		return firstElem;
	};
	
	this.toString = function(){
		if(arguments.length > 0){
			return false;
		}
		for(let i = 0; i < this.size; i++){
			this.str +=this.array[i] + ", ";
		}
		this.str = this.str.substring(0, this.str.length - 2);
		return this.str;
	};

	this.sort = function (f) {
		if(f && (typeof f == "function" )){
			for(let i = 0; i < this.size; i++){
				for(let j = 0; j < this.size; j++){
					if(j === this.size - 1){
						break;
					}
					if(f(this.array[j], this.array[j+1]) === 	1){
						[this.array[j], this.array[j+1]] = [this.array[j+1], this.array[j]];
					}
				}
			}
		}else{
			for(let i = 0; i < this.size; i++){
				for(let j = 0; j < this.size; j++){
					if(j === this.size - 1){
						break;
					}if( String(this.array[j]) > String(this.array[j + 1]) ){
						[this.array[j], this.array[j+1]] = [this.array[j+1], this.array[j]];
					}
				}
			}
		}
	}
}